[Feat] keep track of previously logged out users

aliseyalvi opened the pull request:
### What does this PR?
This PR keep track of previously logged out users and show them in account bottom sheet for easy login. 
User is redirected to login screen with prefilled username

### Issue number
fixes #2904

### Screenshots/Video

https://github.com/user-attachments/assets/7c8236c8-5d90-477a-8594-c5fc2c5665fe

noumantahir commented on src/components/accountsBottomSheet/view/accountsBottomSheetView.js:
```diff
@@ -55,7 +80,7 @@ const AccountsBottomSheet = forwardRef(
         >
           <FlatList
-            data={accounts}
+            data={[...accounts, ...prevLoggedInUsers]}
```
to keep data flow more organised, why not added a section below list instead of mixing different objects? can even add a section labeled `Logged out accounts`

noumantahir commented on src/components/accountsBottomSheet/view/accountsBottomSheetView.js:
```diff
@@ -29,20 +32,42 @@ const AccountsBottomSheet = forwardRef(
-            <Text style={styles.name}>{`@${item.username}`}</Text>
+    const _handlePressAccountTile = (item) => {
+      if (
+        item &&
```
this huge logic can be simplified using `?` and `!` operators.

noumantahir commented on src/components/accountsBottomSheet/view/accountsBottomSheetView.js:
```diff
@@ -29,20 +32,42 @@ const AccountsBottomSheet = forwardRef(
-        )}
-      </TouchableOpacity>
-    );
+          {currentAccount.name === item.username && (
```
There should be an options to remove listed user from logged out users... can add `X` towards end of tile to remove item

noumantahir reviewed (changes requested):
left some review comments, functionality wise, seems good.

aliseyalvi commented:
@noumantahir PR updated according to review

- added option to remove user from list
- added logged out users to separate sections


https://github.com/user-attachments/assets/0981c50d-ba9c-412e-9b43-edef007076b2

aliseyalvi commented:
@noumantahir updated icon
<img width="340" alt="image" src="https://github.com/user-attachments/assets/7323d3e2-9136-433d-9c5a-4b6225aaa1c8">

noumantahir commented on src/redux/reducers/applicationReducer.ts:
```diff
@@ -72,6 +79,7 @@ interface State {
   hidePostsThumbnails: boolean;
   isTermsAccepted: boolean;
   isBiometricEnabled: boolean;
+  prevLoggedInUsers: PrevLoggedInUsers[];
```
not sure if this the right place for `preLoggedInUsers`, for one application store has already overflown, and second since this is more related to account, perhaps it should be in `accountReducer`

<img width="923" alt="Screenshot 2024-08-13 at 15 39 30" src="https://github.com/user-attachments/assets/bb673bd2-6814-42fc-9793-0f4b3de6c3a1">

noumantahir reviewed (changes requested):
One corner case that needs handling is, if I logout of every account and try to login again, there is no way to select from logged out list. it directly lands login screen.

My suggestion would be to open account switch modal at-least at these two places. or only open account switcher if there is at-least one logged out user.

<img width="201" alt="Screenshot 2024-08-13 at 15 44 10" src="https://github.com/user-attachments/assets/d8b78400-16de-4fd3-a25f-3d35abd7cf05">
<img width="197" alt="Screenshot 2024-08-13 at 15 43 57" src="https://github.com/user-attachments/assets/36fd00e7-55df-4d80-b65d-2d7a486b22d0">

aliseyalvi commented:
@noumantahir PR updated as per review.

- show account switch modal if user is logged out
- move prev logged out users data from application store to account store

noumantahir commented:
lgtm
